Quest-3/Part-1/main.py

```diff
@@ -0,0 +1,69 @@
+def main(fileName):
+    lines = readFile(fileName)
+
+    schemEngine = [list(line.replace("\n", "")) for line in lines]
+    symbols = set(char for line in schemEngine for char in line if char != '.' and not char.isdigit())
+
+    allNumbers = findNumbers(schemEngine, symbols)
+    trueNumbers = findSymbolsAroundNumbers(schemEngine, allNumbers, symbols)
+
+    total = sum(trueNumbers)
+    print(total)
+
+def readFile(name):
+    with open(file=f"Quest-3/{name}", mode="r", encoding="utf-8") as f:
+        return f.readlines()
+
+def findNumbers(matrix, symbols):
+    visited = set()
+    numbers = {}
+
+    def exploreNumber(row, col):
+        if row < 0 or col < 0 or row >= len(matrix) or col >= len(matrix[0]) or (row, col) in visited or matrix[row][col] == '.' or matrix[row][col] in symbols:
```

Comment:
which one is wrong?

Reply:
or matrix[row][col] in symbols

Reply:
yeah, that one is weird. You have no idea what symbols could be there. No need to scan for them beforehand. You can check for not being in "." and digits.